rust/kernel/delay.rs

```diff
@@ -56,6 +56,18 @@ pub fn coarse_sleep(duration: Duration) {
     unsafe { bindings::msleep(coarse_sleep_conversion(duration)) }
 }
 
+/// busy wait for enough loop cycles to achieve the desired delay.
+///
+/// This function supports the C side `mdelay`, `udelay`, and `ndelay` functions.
+pub fn coarse_delay(duration: Duration) {
+    let usecs = duration.subsec_micros();
+    if usecs == 0 {
```

Comment:
I don't think this does what you want. If someone specifies a duration that is "1 second + 1 nanosecond" this will return 0 and then we'll sleep for only 1 nanosecond, rather than 1 second and 1 nanosecond.

Reply:
Yeah, fixed. Now the function can handle more than one second. I didn't think that we need to handle busy-loop for more than seconds.